tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -989,3 +997,530 @@ def progress_check():
+                tp_str += "p{} ={:7,.3f}s, ".format(prc[i], tprc[i])
+            # Log them
+            self.logger.debug(tp_str)
+
+        # Prepare command
+        remote_script_path = inject_remote_script(node, "topic_operations.py")
+        cmd = f"python3 {remote_script_path} "
+        cmd += f"--brokers '{brokers}' "
+        cmd += f"--batch-size '{batch_size}' "
+        cmd += "create "
+        cmd += f"--topic-prefix '{topic_name_prefix}' "
+        cmd += f"--topic-count {topic_count} "
+        cmd += "--kafka-batching " if use_kafka_batching else ""
+        cmd += f"--topic-name-length {topic_name_length} "
+        cmd += f"--partitions {num_partitions} "
+        cmd += f"--replicas {num_replicas} "
+        cmd += "--skip-randomize-names" if skip_name_randomization else ""
+        hostname = node.account.hostname
+        self.logger.info(f"Starting topic creation script on '{hostname}")
+        self.logger.debug(f"...cmd: {cmd}")
+
+        data = {}
+        for line in node.account.ssh_capture(cmd):
+            # The script will produce jsons one, per line
+            # And it will have timings only
+            self.logger.debug(f"received {sys.getsizeof(line)}B "
+                              f"from '{hostname}'.")
+            data = self._try_parse_json(node, line.strip())
+            if data is not None:
+                if 'error' in data:
+                    self.logger.warning(f"Node '{hostname}' reported "
+                                        f"error:\n{data['error']}")
+                    raise RuntimeError(data['error'])
+                else:
+                    # Extract data
+                    timings = data.get('timings', {})
+                    log_timings_with_percentiles(timings)
+            else:
+                data = {}
+
+        topic_details = data.get('topics', [])
+        current_count = len(topic_details)
+        self.logger.info(f"Created {current_count} topics")
+        assert len(topic_details) == topic_count, \
+            f"Topic count not reached: {current_count}/{topic_count}"
+
+        return topic_details
+
+    @cluster(num_nodes=10, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_topic_swarm(self):
+        """Test creates 11950 topics, validates partitions and replicas,
+        produces 100 messages to each topic using batches, consumes
+        messages from random topic in each batch and validates message content
+
+        Returns:
+            None
+        """
+
+        # Max number of partitions for i3en.xlarge
+        # default settings is 1000 partitions per shard/cpu
+        # (9 nodes × 4 vcpus/shards × 1000) / 3 replicas
+        # 12000
+
+        # Parameters
+        topic_count = 11950
+        batch_size = 2048
+        topic_name_length = 200
+        num_partitions = 1
+        num_replicas = 3
+        use_kafka_batching = True
+        topic_name_prefix = \
+            f"topic-swarm-create-p{num_partitions}-r{num_replicas}"
```

Comment:
Lets write a python dataclass for the config parameters that this test and `test_many_topics_throughput` share and create a few default config profiles for the tests. One for the values above and another for 40k topics.

Reply:
Based on this, created a series of tickets to work on: https://redpandadata.atlassian.net/browse/PESDLC-886